pkg/flowkit/gateway/emulator.go

```diff
@@ -146,6 +147,20 @@ func (g *EmulatorGateway) GetTransactionResult(tx *flow.Transaction, waitSeal bo
 	return result, nil
 }
 
+// GetTransactionResultByID gets a transaction result by ID from the Flow Access API.
+func (g *EmulatorGateway) GetTransactionResultByID(txID flow.Identifier, waitSeal bool) (*flow.TransactionResult, error) {
+	result, err := g.backend.GetTransactionResult(g.ctx, txID)
+	if err != nil {
+		return nil, UnwrapStatusError(err)
+	}
+
+	if result.Status != flow.TransactionStatusSealed && waitSeal {
+		time.Sleep(time.Second)
+		return g.GetTransactionResultByID(txID, waitSeal)
+	}
+
```

Comment:
I timed it and it takes 1 min + to get transaction since we need to wait for it to be sealed.